### PR TITLE
[SPARK-6428][Streaming] Added explicit types for all public methods.

### PR DESCRIPTION
**Pull request by rxin**

### CODE REVIEW

**Comment by andrewor14** [comment]
@tdas

File: streaming/src/main/scala/org/apache/spark/streaming/Checkpoint.scala [hunk 1]
```diff
@@ -67,12 +67,12 @@ object Checkpoint extends Logging {
   val REGEX = (PREFIX + """([\d]+)([\w\.]*)""").r
 
   /** Get the checkpoint file for the given checkpoint time */
-  def checkpointFile(checkpointDir: String, checkpointTime: Time) = {
+  def checkpointFile(checkpointDir: String, checkpointTime: Time): Path = {
```
Comment (tdas) [inline streaming/src/main/scala/org/apache/spark/streaming/Checkpoint.scala]: This isnt a public method. Are you referring to the fact that in Java bytecode it is public?

Comment (rxin) [inline streaming/src/main/scala/org/apache/spark/streaming/Checkpoint.scala]: it is a def so it is public.

Comment (rxin) [inline streaming/src/main/scala/org/apache/spark/streaming/Checkpoint.scala]: (i.e. yes - "public" here doesn't mean "Spark's public API", but "public" in the sense of java/scala interfaces)

File: streaming/src/main/scala/org/apache/spark/streaming/dstream/DStream.scala [hunk 2]
```diff
@@ -619,14 +619,16 @@ abstract class DStream[T: ClassTag] (
...
   def print(num: Int) {
-    def foreachFunc = (rdd: RDD[T], time: Time) => {
-      val firstNum = rdd.take(num + 1)
-      println ("-------------------------------------------")
-      println ("Time: " + time)
-      println ("-------------------------------------------")
-      firstNum.take(num).foreach(println)
-      if (firstNum.size > num) println("...")
-      println()
+    def foreachFunc: (RDD[T], Time) => Unit = {
```
Comment (tdas) [inline streaming/src/main/scala/org/apache/spark/streaming/dstream/DStream.scala]: Why this?

Comment (rxin) [inline streaming/src/main/scala/org/apache/spark/streaming/dstream/DStream.scala]: the current version of the linter has a bug that it thinks inline defs as public defs.

Comment (tdas) [inline streaming/src/main/scala/org/apache/spark/streaming/dstream/DStream.scala]: :/

**Comment by tdas** [comment]
LGTM.

**Comment by rxin** [comment]
Thanks. Merging in master.

**Pull request closed without merging**